omnigibson/prims/prim_base.py

```diff
@@ -71,26 +71,39 @@ def initialize(self):
         """
         assert (
             not self._initialized
-        ), f"Prim {self.name} at prim_path {self._prim_path} can only be initialized once! (It is already initialized)"
+        ), f"Prim {self.name} at prim_path {self.prim_path} can only be initialized once! (It is already initialized)"
         self._initialize()
 
-        # Cache state size
-        self._state_size = len(self.dump_state(serialized=True))
-
         self._initialized = True
 
-    def load(self):
+    def load(self, scene):
         """
         Load this prim into omniverse, and return loaded prim reference.
 
         Returns:
             Usd.Prim: Prim object loaded into the simulator
         """
-        if self._loaded:
-            raise ValueError(f"Cannot load prim {self.name} multiple times.")
+        # Load the prim if it doesn't exist yet.
+        assert (
+            not self._loaded
+        ), f"Prim {self.name} at prim_path {self.prim_path} can only be loaded once! (It is already loaded)"
+
+        # Assign the scene first.
+        self._scene = scene
+        self._scene_assigned = True
+
+        # Then check if the prim is already loaded
+        if lazy.omni.isaac.core.utils.prims.is_prim_path_valid(prim_path=self.prim_path):
+            log.debug(f"prim {self.name} already exists, skipping load")
+            self._prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(prim_path=self.prim_path)
+        else:
+            # If not, we'll load it.
+            # Override w/e value is available via created_manually - we are definitely creating this
+            # object manually now!
+            self._created_manually = True
+            self._prim = self._load()
```

Comment:
when would this else statement happen but `self._created_manually` was originally set to False?

Reply:
Not sure but removed.